tests/FunctionalJavascript/TranslationSynchronisationFieldWidgetTest.php

```diff
@@ -0,0 +1,164 @@
+      'weight' => 1,
+      'region' => 'content',
+      'type' => 'oe_translation_translation_sync_widget',
+      'settings' => [],
+      'third_party_settings' => [],
+    ]);
+    $entity_form_display->save();
+
+    $user = $this->drupalCreateUser([], NULL, TRUE);
+    $this->drupalLogin($user);
+  }
+
+  /**
+   * Tests the Translation Synchronisation field widget.
+   */
+  public function testTranslationSynchronisationFieldWidget(): void {
+    $this->drupalGet('/node/add/page');
+
+    $assert = $this->assertSession();
+    $assert->fieldExists('title[0][value]')->setValue('My page');
+
+    // Assert the select is correct.
+    $select = $assert->selectExists('translation_sync[0][type]');
+    $assert->optionExists('Select type', 'None');
+    $assert->optionExists('Select type', 'Manual');
+    $assert->optionExists('Select type', 'Automatic with minimum threshold');
+    $this->assertEquals('None', $select->find('xpath', "//option[@selected='selected']")->getText());
+
+    // Assert we don't see a configuration fieldset.
+    $this->assertFalse($this->getSession()->getPage()->find('css', '#edit-translation-sync-0-configuration')->isVisible());
+
+    // Save the node with no value.
+    $this->getSession()->getPage()->pressButton('Save');
+    /** @var \Drupal\node\NodeInterface $node */
+    $node = $this->entityTypeManager->getStorage('node')->load(1);
+    $this->assertTrue($node->get('translation_sync')->isEmpty());
+    $this->assertEmpty($node->get('translation_sync')->getValue());
+
+    // Edit the node again and configure it to be Manual.
+    $this->drupalGet($node->toUrl('edit-form'));
+    $this->getSession()->getPage()->selectFieldOption('Select type', 'manual');
+    $this->getSession()->wait(10);
+    $this->assertFalse($this->getSession()->getPage()->find('css', '#edit-translation-sync-0-configuration')->isVisible());
+    $this->getSession()->getPage()->pressButton('Save');
+    $this->entityTypeManager->getStorage('node')->resetCache();
+    $node = $this->entityTypeManager->getStorage('node')->load(1);
+    $expected_values = [
+      'type' => 'manual',
+      'configuration' => [],
+    ];
+    $this->assertEquals($expected_values, $node->get('translation_sync')->first()->getValue());
+
+    // Edit the node again and configure it to be Automatic without a date.
+    $this->drupalGet($node->toUrl('edit-form'));
+    $this->getSession()->getPage()->selectFieldOption('Select type', 'automatic');
+    $assert->waitForField('Languages');
+    $this->assertTrue($this->getSession()->getPage()->find('css', '#edit-translation-sync-0-configuration')->isVisible());
+    $this->assertTrue($this->getSession()->getPage()->find('css', '#edit-translation-sync-0-configuration-languages')->isVisible());
+    $this->assertTrue($this->getSession()->getPage()->find('css', '#edit-translation-sync-0-configuration-date')->isVisible());
+    $this->getSession()->getPage()->selectFieldOption('Languages', 'en', TRUE);
```

Comment:
We could also check that the languages field is required but the date field is not.

Reply:
Ah yes, true

Reply:
It's ok, we cover this in the kernel